commit 2aeedfb039d67e3348eb86f8031933d0a7dd3e8b (refs/changes/35/335/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-08-23T14:50:02-07:00 (3 years, 2 months ago)
    
    MANTA-2966 binder should check names for validity before passing to ZK client

diff --git a/lib/server.js b/lib/server.js
index d9cc2d7..6821828 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -130,6 +130,16 @@ function resolve(options, query, cb) {
                 return;
         }
 
+        domain = domain.toLowerCase();
+        if (/[^a-z0-9_.-]/.test(domain)) {
+                req.log.debug('request for an invalid name: this client is ' +
+                    'probably misbehaving');
+                query.setError('eserver');
+                query.respond();
+                cb();
+                return;
+        }
+
         req.domain = domain;
         zk.resolveName(req, function (err, record) {
                 if (err && (typeof (err.getCode) !== 'function' ||
